### PR TITLE
Add Support for CORS Headers to the Webhook Source

### LINKED ISSUE #41: Add support for CORS headers
I received a request for the ability have the HTTP Source send CORS headers during preflight requests where the invoker may require the header for verification.

### PR DESCRIPTION
**Pull request by JeffNeff**
This PR attaches a CORS response header to the response of the incoming POST request. 

If the user specifies the `corsOrigin` option in the spec, upon every request we will now attach the following header parameter: 

```
"Access-Control-Allow-Origin" : <spec.corsOrigin>
```

Closes #41

### CODE REVIEW

File: config/300-webhooksource.yaml [hunk 1]
```diff
@@ -57,6 +57,9 @@ spec:
                   context in which an event happened. Must be expressed as a URI-reference. Please refer to the
                   CloudEvents specification for more details: https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#source-1"
                 type: string
+              corsOrigin:
```
Comment (cab105) [inline config/300-webhooksource.yaml:60]: I'm sorry, but the name should reflect the full meaning of the header. How about `corsAllowOrigin`?

**Review by cab105 — CHANGES_REQUESTED** [review]
Thank you for the PR. I don't see anything from a logic perspective, but would prefer a name to reflect the full header being set as AllowOrigin. This is to reduce conflict in the other CORS header that is sent by the browser called `Origin`.

File: pkg/sources/adapter/webhooksource/webhook.go [hunk 2]
```diff
@@ -98,6 +97,10 @@ func runHandler(ctx context.Context, s *http.Server) error {
 // handleAll receives all webhook events at a single resource, it
 // is up to this function to parse event wrapper and dispatch.
 func (h *webhookHandler) handleAll(w http.ResponseWriter, r *http.Request) {
+	if h.corsAllowOrigin != "" {
```
Comment (JeffNeff) [inline pkg/sources/adapter/webhooksource/webhook.go:100]: @cab105 is this right? Or should this be checking aginst a `nil`?

Comment (cab105) [reply pkg/sources/adapter/webhooksource/webhook.go:100]: Much better, thank you!

File: config/300-webhooksource.yaml (hunk 1 above)
Comment (cab105) [reply config/300-webhooksource.yaml]: Thank you!

File: pkg/sources/adapter/webhooksource/webhook.go (hunk 2 above)
Comment (JeffNeff) [reply pkg/sources/adapter/webhooksource/webhook.go:100]: okay! thank you

**Review by cab105 — APPROVED** [review]

**Pull request merged**